pkg/cmd/node/node_diagnostics.go

```diff
@@ -0,0 +1,55 @@
+package node
+
+import (
+	"github.com/dcos/dcos-cli/api"
+	"github.com/spf13/cobra"
+)
+
+// newCmdNodeMetrics creates the `core node metrics` subcommand.
+func newCmdNodeDiagnostics(ctx api.Context) *cobra.Command {
+	var cancel bool
+	var jsonOutput bool
+	var list bool
+	var status bool
+	cmd := &cobra.Command{
+		Use:   "diagnostics",
+		Short: "Use diagnostics bundles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var subCommand *cobra.Command
+			switch {
+			case cancel && !list && !status:
+				subCommand = newCmdNodeDiagnosticsCancel(ctx)
+			case !cancel && list && !status:
+				subCommand = newCmdNodeDiagnosticsList(ctx)
+			case !cancel && !list && status:
```

Comment:
It is harder to read when checking for mutual exclusion, usually when we do this we also print an informative error message. Given that it is also deprecated we should keep it simple, here I'd just check for each option individually and match the first one : 
``` go
if list {
    subCommand = newCmdNodeDiagnosticsList(ctx)
} else if status {
    // ...
} // ...
```